src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -362,6 +363,12 @@ internal BuildData GetMaestroBuildDataFromMergedManifest(
 
             foreach (var package in buildModel.Artifacts.Packages)
             {
+                if (package.Attributes.TryGetValue(VisibilityAttributeName, out string visibility)
```

Comment:
With this change, it should not be necessary to skip publishing in `src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifestBase.cs`

Reply:
This is required as far as I can tell, because the code in `PublishArtifactsInManifestBase` reads from the asset manifests, not from the BAR records, and the merged manifest (or non-merged if there's only one) can still have non-external visibility.

Reply:
The goal would be to remove the non-external artifacts from the merged manifest.

Reply:
I didn't really see much reason to remove them from the merged manifest if repos with only one manifest wouldn't go down this path anyway.

I'm exploring an alternative solution as mentioned in https://github.com/dotnet/arcade/pull/15344#discussion_r1890885830 that should obsolete this PR.